src/Query/SqlWalker.php

```diff
@@ -1412,7 +1412,7 @@ public function walkObjectExpression(string $dqlAlias, array $partialFieldSet, s
 
             $sqlParts[] = $col . ' AS ' . $columnAlias;
 
-            $this->scalarResultAliasMap[$resultAlias][] = $columnAlias;
+            $this->scalarResultAliasMap[$resultAlias ?? ''][] = $columnAlias;
```

Comment:
should it even be added in `scalarResultAliasMap` if there is no result alias ?

Reply:
after re-reading how the map is accessed in `walkResultVariable()` it's probably safe to assume that adding an entry here in case of `$resultAlias` being `null` doesn't make much sense